### PR TITLE
More info on failure to parse dataProvider

### PR DESCRIPTION
**Pull request by sh41**
Provide exception file location, line number and message when thrown during dataProvider execution.

An example failure: Missing extension for an included module. 
Current output: 
```
  [Codeception\Exception\TestParseException]
  Couldn't parse test '/var/www/html/tests/Unit/TestCest.php'
  DataProvider 'myProvider' for TestCest->testSomeThings is invalid or not callable.
  Make sure that the dataprovider exist within the test class.
```
New output:
```
  [Codeception\Exception\TestParseException]
  Couldn't parse test '/var/www/html/tests/Unit/TestCest.php'
  DataProvider 'myProvider' for TestCest->testSomeThings is invalid or not callable.
  Make sure that the dataprovider exist within the test class.
  /var/www/html/vendor/phpseclib/phpseclib/phpseclib/Math/BigInteger.php:90:GMP is not setup correctly on this system
```

### CODE REVIEW

**Comment by Naktibalda** [comment]
In your example additional message provides conflicting information with the main message.

Would it be possible to replace catch(Exception) with more specific catch matching only invalid data providers or implement a better validation?

File: tests/cli/RunCest.php [hunk 1]
```diff
@@ -190,6 +190,57 @@ public function runTestWithDataProviders(\CliGuy $I)
...
+    public function runTestWithDataProvidersExceptionStdout(\CliGuy $I)
+    {
+        $I->executeCommand('run tests/dummy/DataProvidersExceptionCest.php 2> /dev/null', false);
+        $I->dontSeeInShellOutput('There was 1 warning');
+        $I->dontSeeInShellOutput('PHPUnit_Framework_Warning');
+        $I->dontSeeInShellOutput('The data provider specified for DataProvidersExceptionTest::testIsTriangle is invalid.');
+        $I->dontSeeInShellOutput("WARNINGS!");
+        $I->dontSeeInShellOutput("OK");
+        $I->dontSeeInShellOutput("[Exception]");
+        $I->dontSeeInShellOutput("Something went wrong!!!");
```
Comment (Naktibalda) [inline tests/cli/RunCest.php]: This test is very negative, could you add some `seeInShellOutput` assertions to it?

File: tests/data/claypit/tests/dummy/DataProvidersExceptionCest.php [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+    public function testIsTriangle(DumbGuy $I)
+    {
+        $I->amGoingTo("Fail with an exception before I even get here");
+    }
+
+    // The test of this relies upon the line numbers being unchanged. If you do need to add lines
+    // please change the relevant test in tests/cli/RunCest:runTestWithDataProvidersExceptionStderrVerbose
+    public function triangles()
+    {
+        throw new \Exception("Something went wrong!!!");
```
Comment (Naktibalda) [inline tests/data/claypit/tests/dummy/DataProvidersExceptionCest.php]: Oh, this test broke a bunch of other tests in cli suite.
Please find a different place for it.

File: tests/cli/RunCest.php (hunk 1 above)
Comment (sh41) [inline tests/cli/RunCest.php]: This test deliberately redirects StdErr which is where all the good stuff appears, the StdOut only looks like this:
```
$ codecept run tests/failure_exceptions/DataProvidersExceptionCest.php 2> /dev/null
Codeception PHP Testing Framework v2.3.4
Powered by PHPUnit 5.7.21 by Sebastian Bergmann and contributors.

$
```
I shied away from testing for the header/versions as they are likely to change. Could add a test for "Codeception" and "Powered by PHPUnit"?

Comment (sh41) [inline tests/cli/RunCest.php]: Actually - the output from the `executeCommand` strips even the 
```
Codeception PHP Testing Framework v2.3.4
Powered by PHPUnit 5.7.21 by Sebastian Bergmann and contributors.
```
headers - I've added a test to check that the output is blank and the result code is 1.

Comment (Naktibalda) [inline tests/cli/RunCest.php]: I think that the name of this test isn't self-documenting enough, could you add this explanation as a comment?

File: tests/data/bundled_suites/_support/_generated/UnitTesterActions.php [hunk 3]
```diff
@@ -0,0 +1,547 @@
+<?php  //[STAMP] db3bd0913ded92ae845d337ddebb7e71
```
Comment (Naktibalda) [inline tests/data/bundled_suites/_support/_generated/UnitTesterActions.php]: Please delete this file from git.
I've raised a pull request to fix gitignore rules https://github.com/Codeception/Codeception/pull/4439

**Comment by sh41** [comment]
Apologies for all of the commits - I seem to get very different results from the CI environments than I do on my dev box. Is there a way of running the CI tests without having to push to github each time?

**Comment by sh41** [comment]
I'm going to rethink how these tests are being executed. Their inclusion in the claypit with all the other tests seems to be causing a lot of knock on trouble. 

The latest issue is with the TestLoaderTest routine which attempts to load all tests in the claypit. The exception being thrown by one of the new tests is causing issues with loading and as far as I can see there is no way to sensibly exclude a folder from the routine being tested. 

Do you have any suggestions @Naktibalda? I am thinking that I will look at refactoring into a different directory so that they do not get included by other tests that pick up the entire claypit suite?

**Comment by Naktibalda** [comment]
Don't put your tests to claypit (I think that it is only necessary for testing generator commands),
put your test or suite to `tests/data`

**Comment by sh41** [comment]
Have reverted all changes made to claypit/RunCest and refactored the sub tests into a separate suite and tests and added a new Cest separate from RunCest to handle execution of them. Hopefully this will provide appropriate separation so that the execution doesn't effect other suites/tests.

**Comment by sh41** [comment]
Travis build has failed, with this message:
```
Failed to clone the git@github.com:twitter/typeahead.js.git repository, try running in interactive mode so that you can enter your GitHub credentials
                                                                                                                                        
  [Composer\Repository\InvalidRepositoryException]                                                                                      
  No valid bower.json was found in any branch or tag of https://github.com/twitter/typeahead.js.git, could not load a package from it.  
                                                                                                                                        
```

I don't think I've caused this? Is there something I need to do to get Travis to pass?

**Comment by Naktibalda** [comment]
It is probably `yii-basic-app` causing issues again.
I restarted the build before checking, but I will look at the output if it fails again.

**Comment by Naktibalda** [comment]
@samdark yii2 is breaking Codeception builds again.

```
Installing yiisoft/yii2-app-basic (2.0.12)
  - Installing yiisoft/yii2-app-basic (2.0.12): Downloading (100%)
Created project in frameworks-yii-basic
Loading composer repositories with package information
Updating dependencies
Failed to clone the git@github.com:jquery/jquery-dist.git repository, try running in interactive mode so that you can enter your GitHub credentials
                                                                                                                                      
  [Composer\Repository\InvalidRepositoryException]                                                                                    
  No valid bower.json was found in any branch or tag of https://github.com/jquery/jquery-dist.git, could not load a package from it. 
```

**Comment by DavertMik** [comment]
Wow, thanks for the work and so many nice tests! Finally we got CI issues sorted out so I think this PR can be merged

**Pull request merged**